### PR TITLE
Upgrade receptionist

### PR DESCRIPTION
**Pull request by GustavoDCC**
Picovoice model updated to detect person names

### CODE REVIEW

File: challenge_receptionist/src/challenge_receptionist/learn_guest.py [hunk 1]
```diff
@@ -134,14 +133,10 @@ def __init__(self, robot,
             else:
                 smach.StateMachine.add('HEAR_DRINK_ANSWER',
                                        HearOptionsExtraPicoVoice(
-                                           robot, "receptionist", answer.writeable, "drinks", look_at_standing_person=True),
-                                       transitions={'heard': 'PROCESS_ANSWER',
+                                           robot, "receptionist", guest_drink_des.writeable, ["drinks"], look_at_standing_person=True),
```
Comment (MatthijsBurgh) [inline challenge_receptionist/src/challenge_receptionist/learn_guest.py:136]: This argument needs to be a designator resolving to a HMIResult......

Comment (GustavoDCC) [reply challenge_receptionist/src/challenge_receptionist/learn_guest.py:136]: This argument is already a designator resolving to a HMIResult coming from HandleSingleGuest - receptionist.py

**Review by MatthijsBurgh — APPROVED** [review]

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
Fix the mess of `process_answer`

File: robot_smach_states/src/robot_smach_states/human_interaction/human_interaction.py [hunk 2]
```diff
@@ -600,6 +600,20 @@ def execute(self, userdata=None):
...
+def process_answer(_, context: Union[ds.Designator[str], str], answer_des, output_des):
+    try:
+        answer_val = answer_des.resolve()
+        rospy.logdebug(f"{answer_val=}")
+        semantic = answer_val.semantics[ds.value_or_resolve(context)]
+        rospy.loginfo(f"Your answer is: '{semantic}'")
+        output_des.write(semantic)
+
+    except KeyError as e:
+        rospy.loginfo(f"KeyError resolving the name heard: {e}")
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/human_interaction/human_interaction.py:613]: Name is not correct here

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

**Pull request merged**